main.tf

```diff
@@ -3,7 +3,7 @@ locals {
   aws_azs = slice(data.aws_availability_zones.available.names[*], 0, var.number_of_azs)
 
   # calculate VPC and subnets CIDRs
-  vpc_cidr         = cidrsubnet(data.aws_ssm_parameter.network_range.value, 5, var.env_index)
+  vpc_cidr         = "${ var.lookup_ssm_param == false ? cidrsubnet(data.aws_ssm_parameter.network_range.value, 5, var.env_index) : cidrsubnet(data.aws_ssm_parameter.network_range_per_app[0].value, 5, var.env_index) }"
```

Comment:
is there no way to have an empty data source ? (without using the lookup var)
To make the code more clean and readable I would also have this condition as a separate line for setting the vpc_cidr and then do the cidrsubset function